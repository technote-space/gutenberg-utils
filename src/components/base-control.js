const { BaseControl } = wp.components;

import classnames from 'classnames';

if ( ! BaseControl.VisualLabel ) {
	BaseControl.VisualLabel = ( { className, children } ) => {
		className = classnames( 'components-base-control__label', className );
		return (
			<span className={ className }>
				{ children }
			</span>
		);
	};
}

export default BaseControl;
