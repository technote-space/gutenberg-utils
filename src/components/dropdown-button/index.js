import React from 'react';
import classnames from 'classnames';
import {Dropdown, IconButton} from '@wordpress/components';

/**
 * @param {*} icon icon
 * @param {string|*} label label
 * @param {string|*} tooltip tooltip
 * @param {string} className class name
 * @param {string} position position
 * @param {string} contentClassName content class name
 * @param {function} renderContent render content
 * @param {boolean} isActive is active?
 * @param {boolean} isDisabled is disabled?
 * @param {boolean} isDropdownDisabled is dropdown disabled?
 * @param {boolean} isHiddenIndicator is hidden indicator?
 * @returns {*} dropdown button
 * @constructor
 */
const DropdownButton = ({
  icon = 'menu',
  label,
  tooltip = label,
  className,
  position = 'top right',
  contentClassName,
  renderContent,
  isActive = false,
  isDisabled = false,
  isDropdownDisabled = false,
  isHiddenIndicator = false,
}) => {
  return <Dropdown
    className={classnames('components-dropdown-toggle', className)}
    contentClassName={classnames('components-dropdown-button__popover', contentClassName)}
    position={position}
    renderToggle={({isOpen, onToggle, onClose}) => {
      if (isOpen && isDropdownDisabled) {
        // eslint-disable-next-line no-magic-numbers
        setTimeout(onClose, 0);
      }

      return <IconButton
        className={classnames('components-dropdown-button__toggle', 'components-toolbar__control', 'utils--components-toolbar__control', {
          'is-active is-pressed': isActive,
        })}
        icon={icon}
        onClick={onToggle}
        aria-haspopup="true"
        aria-expanded={isOpen}
        aria-pressed={isActive}
        label={label}
        tooltip={tooltip}
        disabled={isDisabled}
      >
        {!isHiddenIndicator && <span className="components-dropdown-menu__indicator"/>}
      </IconButton>;
    }}
    renderContent={renderContent}
  />;
};

export default DropdownButton;
