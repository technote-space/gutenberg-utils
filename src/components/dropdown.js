/**
 * Gutenberg v6.2
 */
import Popover from './popover';

const { Component, createRef } = wp.element;

/**
 * Dropdown
 */
class Dropdown extends Component {

	constructor() {
		super( ...arguments );

		this.toggle = this.toggle.bind( this );
		this.close = this.close.bind( this );
		this.closeIfClickOutside = this.closeIfClickOutside.bind( this );

		this.containerRef = createRef();
		this.state = {
			isOpen: false,
		};
	}

	componentWillUnmount() {
		const { isOpen } = this.state;
		const { onToggle } = this.props;
		if ( isOpen && onToggle ) {
			onToggle( false );
		}
	}

	componentDidUpdate( prevProps, prevState ) {
		const { isOpen } = this.state;
		const { onToggle } = this.props;
		if ( prevState.isOpen !== isOpen && onToggle ) {
			onToggle( isOpen );
		}
	}

	toggle() {
		this.setState( state => ( {
			isOpen: ! state.isOpen,
		} ) );
	}

	/**
	 * @param {Event} event event
	 */
	closeIfClickOutside( event ) {
		if ( ! this.containerRef.current.contains( event.target ) ) {
			const target = document.querySelector( '.components-color-picker, .components-font-size-picker__dropdown-content' );
			if ( target && target.contains( event.target ) ) {
				return;
			}
			this.close();
		}
	}

	close() {
		this.setState( { isOpen: false } );
	}

	render() {
		const { isOpen } = this.state;
		const {
			renderContent,
			renderToggle,
			position = 'bottom',
			className,
			contentClassName,
			expandOnMobile,
			headerTitle,
			focusOnMount,
			popoverProps,
		} = this.props;

		const args = { isOpen, onToggle: this.toggle, onClose: this.close };

		return (
			<div className={ className } ref={ this.containerRef }>
				{ renderToggle( args ) }
				{ isOpen && (
					<Popover
						className={ contentClassName }
						position={ position }
						onClose={ this.close }
						onClickOutside={ this.closeIfClickOutside }
						expandOnMobile={ expandOnMobile }
						headerTitle={ headerTitle }
						focusOnMount={ focusOnMount }
						{ ...popoverProps }
					>
						{ renderContent( args ) }
					</Popover>
				) }
			</div>
		);
	}
}

export default Dropdown;
