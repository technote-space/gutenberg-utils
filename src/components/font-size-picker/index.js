/**
 * Gutenberg v6.1
 **/
const { map } = window.lodash;
const { __, _x, sprintf } = wp.i18n;
const { Dashicon, Button, RangeControl, NavigableMenu } = wp.components;
import Dropdown from '../dropdown';
import BaseControl from '../base-control';
import './style.scss';


/**
 * @param {*} fallbackFontSize fallback font size
 * @param {Array} fontSizes font sizes
 * @param {boolean} disableCustomFontSizes disable custom font sizes
 * @param {function} onChange on change
 * @param {*} value value
 * @param {boolean} withSlider with slider
 * @return {null|Component} font size picker
 */
function FontSizePicker({
	fallbackFontSize,
	fontSizes = [],
	disableCustomFontSizes = false,
	onChange,
	value,
	withSlider = false,
}) {
	if (disableCustomFontSizes && !fontSizes.length) {
		return null;
	}
	const onChangeValue = (event) => {
		const newValue = event.target.value;
		if (newValue === '') {
			onChange(undefined);
			return;
		}
		onChange(Number(newValue));
	};

	const currentFont = fontSizes.find((font) => font.size === value);
	const currentFontSizeName = (currentFont && currentFont.name) || (!value && _x('Normal', 'font size name')) || _x('Custom', 'font size name');

	return (
		<BaseControl>
			<BaseControl.VisualLabel>
				{__('Font Size')}
			</BaseControl.VisualLabel>
			<div className="components-font-size-picker__buttons">
				{fontSizes.length &&
				<Dropdown
					className="components-font-size-picker__dropdown"
					contentClassName="components-font-size-picker__dropdown-content"
					position="bottom"
					renderToggle={({ isOpen, onToggle }) => (
						<Button
							className="components-font-size-picker__selector"
							isLarge
							onClick={onToggle}
							aria-expanded={isOpen}
							aria-label={sprintf(
								/* translators: %s: font size name */
								__('Font size: %s'), currentFontSizeName,
							)}
						>
							{currentFontSizeName}
						</Button>
					)}
					renderContent={() => (
						<NavigableMenu>
							{map(fontSizes, ({ name, size, slug }) => {
								const isSelected = (value === size || (!value && slug === 'normal'));

								return (
									<Button
										key={slug}
										onClick={() => onChange(slug === 'normal' ? undefined : size)}
										className={`is-font-${slug}`}
										role="menuitemradio"
										aria-checked={isSelected}
									>
										{isSelected && <Dashicon icon="saved" />}
										<span className="components-font-size-picker__dropdown-text-size" style={{ fontSize: size }}>
											{name}
										</span>
									</Button>
								);
							})}
						</NavigableMenu>
					)}
				/>
				}
				{(!withSlider && !disableCustomFontSizes) &&
				<input
					className="components-range-control__number"
					type="number"
					onChange={onChangeValue}
					aria-label={__('Custom font size')}
					value={value || ''}
				/>
				}
				<Button
					className="components-color-palette__clear"
					type="button"
					disabled={value === undefined}
					onClick={() => onChange(undefined)}
					isSmall
					isDefault
				>
					{__('Reset')}
				</Button>
			</div>
			{withSlider &&
			<RangeControl
				className="components-font-size-picker__custom-input"
				label={__('Custom Size')}
				value={value || ''}
				initialPosition={fallbackFontSize}
				onChange={onChange}
				min={12}
				max={100}
				beforeIcon="editor-textcolor"
				afterIcon="editor-textcolor"
			/>
			}
		</BaseControl>
	);
}

export default FontSizePicker;
