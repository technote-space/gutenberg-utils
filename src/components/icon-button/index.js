import { comparePackageVersion } from '../../helpers/compatibility';

/**
 * @param {object} props props
 * @return {null|Component} icon button
 */
function IconButton(props) {
	let Component;

	/* istanbul ignore next */
	if (comparePackageVersion('wp-components', '8.5.0', '>=')) {
		Component = require('./implement').default;
	} else {
		/* istanbul ignore next */
		Component = wp.components.IconButton;
	}

	return (
		<Component {...props} />
	);
}

export default IconButton;
