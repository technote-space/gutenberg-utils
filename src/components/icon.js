import classnames from 'classnames';

const {SVG, Dashicon} = wp.components;
const {isURL}         = wp.url;

const {isString} = window.lodash;
import {ICON_SIZE} from '../constant';

const isSVG      = icon => isString(icon) && (isURL(icon) || /^data:image/.test(icon));
const isDashicon = icon => isString(icon) && /^dashicons-/.test(icon);

const createSVG      = (icon, className) => <SVG
  version="1.1"
  xmlns="http://www.w3.org/2000/svg"
  x="0px"
  y="0px"
  width={`${ICON_SIZE}px`}
  height={`${ICON_SIZE}px`}
  viewBox={`0 0 ${ICON_SIZE} ${ICON_SIZE}`}
  enableBackground={`new 0 0 ${ICON_SIZE} ${ICON_SIZE}`}
  className={className}
>
  <image
    width={ICON_SIZE}
    height={ICON_SIZE}
    x="0"
    y="0"
    xlinkHref={icon}
  />
</SVG>;
const createDashicon = (icon, className) => <Dashicon
  icon={icon.replace(/^dashicons-/, '')}
  className={className}
/>;

/**
 * @param {*} icon icon
 * @param {string} className class name
 * @param {*} defaultIcon default icon
 * @returns {null|*|null|*} icon
 * @constructor
 */
const Icon = (
  {
    icon,
    className,
    defaultIcon = false,
  },
) => {
  if (isSVG(icon)) {
    return createSVG(icon, classnames('components-icon components-icon__svg', className));
  }

  if (isDashicon(icon)) {
    return createDashicon(icon, classnames('components-icon components-icon__dashicon', className));
  }

  if (!icon && defaultIcon) {
    return Icon({
      icon: defaultIcon,
      className: className,
    });
  }

  if (!icon) {
    return null;
  }

  return icon;
};

export default Icon;
