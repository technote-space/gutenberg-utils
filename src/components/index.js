import Icon from './icon';
import DropdownButton from './dropdown-button';
import ColorPalette from './color-palette';
import FontSizePicker from './font-size-picker';

export { Icon, DropdownButton, ColorPalette, FontSizePicker };
