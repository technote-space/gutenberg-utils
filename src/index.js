import * as Components from './components';
import * as Helpers from './helpers';

Helpers.applyStyles('.components-dropdown-button__popover .components-font-size-picker__controls .components-custom-select-control__menu', {
	position: 'fixed',
	minWidth: '150px',
});
Helpers.applyStyles('.components-dropdown-button__popover .components-popover__content', {
	padding: '12px',
});
Helpers.applyStyles('.components-dropdown-button__has-property-color .dashicon', {
	color: '#d94f4f',
});
Helpers.applyStyles('.components-dropdown-button__has-property-background-color .dashicon', {
	backgroundColor: '#f0b849',
});
Helpers.applyStyles('.components-dropdown-toggle>.components-toolbar__control.components-button', {
	width: 'auto',
});
Helpers.applyStyles('.components-dropdown-toggle>.components-toolbar__control.components-button.has-icon.has-text>svg', {
	margin: 0,
});

export { Components, Helpers };
