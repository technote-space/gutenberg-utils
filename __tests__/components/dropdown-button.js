/* eslint-disable no-magic-numbers */
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DropdownButton } from '../../src/components';
import Dropdown from '../../src/components/dropdown-button/dropdown';

const { FontSizePicker, ColorPalette } = wp.components;

describe( 'DropdownButton', () => {
	const getSnapshotName = ( name, index ) => `${ name }--${ index }`;

	[
		{
			props: {
				icon: 'dashicons-admin-site',
				label: 'test1-label',
				tooltip: 'test1-tooltip',
				className: 'test1-class',
				position: 'bottom left',
				contentClassName: 'test1-content-class',
				renderContent: () => <div className={ 'popover1' }>test1</div>,
				isActive: true,
				isDisabled: false,
				isHiddenIndicator: true,
			},
			callback: ( wrapper, index ) => {
				expect( wrapper.find( '.test1-class' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.test1-class button' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.test1-class button' ).hostNodes().prop( 'disabled' ) ).toBeFalsy();
				expect( wrapper.find( '.test1-class button.is-active' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.test1-class button .components-dropdown-button__indicator' ).hostNodes() ).toHaveLength( 0 );
				expect( wrapper.find( '.popover1' ).hostNodes() ).toHaveLength( 0 );

				wrapper.find( '.test1-class button' ).hostNodes().simulate( 'click' );

				expect( toJson( wrapper, { mode: 'deep' } ) ).toMatchSnapshot( getSnapshotName( 'open', index ) );

				expect( wrapper.find( '.popover1' ).hostNodes() ).toHaveLength( 1 );

				wrapper.setProps( { isDropdownDisabled: true } );
				wrapper.update();

				expect( wrapper.find( '.popover1' ).hostNodes() ).toHaveLength( 0 );
			},
		},
		{
			props: {
				label: 'test2-label',
				className: 'test2-class',
				contentClassName: 'test2-content-class',
				renderContent: () => <FontSizePicker
					fontSizes={ [
						{ name: 'test1', size: '10px', slug: 'test1' },
						{ name: 'test2', size: '20px', slug: 'test2' },
						{ name: 'test3', size: '30px', slug: 'test3' },
					] }
					value={ 20 }
					fallbackFontSize={ 20 }
					onChange={ () => null }
				/>,
			},
			callback: ( wrapper, index ) => {
				expect( wrapper.find( '.test2-class' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.test2-class button' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.test2-class button' ).hostNodes().prop( 'disabled' ) ).toBeFalsy();
				expect( wrapper.find( '.test2-class button.is-active' ).hostNodes() ).toHaveLength( 0 );
				expect( wrapper.find( '.test2-class button .components-dropdown-button__indicator' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.components-select-control__input' ).hostNodes() ).toHaveLength( 0 );

				{
					wrapper.find( '.test2-class .components-dropdown-button__toggle' ).hostNodes().simulate( 'click' );
					expect( toJson( wrapper, { mode: 'deep' } ) ).toMatchSnapshot( getSnapshotName( 'open', index ) );
					expect( wrapper.find( '.components-select-control__input' ).hostNodes() ).toHaveLength( 1 );

					const outerNode = document.createElement( 'div' );
					document.body.appendChild( outerNode );
					outerNode.dispatchEvent( new Event( 'click' ) );
					wrapper.update();

					expect( wrapper.find( '.components-select-control__input' ).hostNodes() ).toHaveLength( 0 );
				}
			},
		},
		{
			props: {
				label: 'test3-label',
				className: 'test3-class',
				contentClassName: 'test3-content-class',
				renderContent: () => <div className={ 'popover3' }>test3</div>,
				isActive: false,
				isDisabled: true,
			},
			callback: ( wrapper ) => {
				expect( wrapper.find( '.test3-class' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.test3-class button' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.test3-class button' ).hostNodes().prop( 'disabled' ) ).toBeTruthy();
				expect( wrapper.find( '.test3-class button.is-active' ).hostNodes() ).toHaveLength( 0 );
				expect( wrapper.find( '.test3-class button .components-dropdown-button__indicator' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.popover3' ).hostNodes() ).toHaveLength( 0 );
			},
		},
		{
			props: {
				label: 'test4-label',
				className: 'test4-class',
				contentClassName: 'test4-content-class',
				renderContent: () => <ColorPalette
					colors={ [
						{ name: 'test1', color: 'red', slug: 'test1' },
						{ name: 'test2', color: 'green', slug: 'test2' },
						{ name: 'test3', color: 'blue', slug: 'test3' },
					] }
					value={ 'green' }
					disableCustomColors={ false }
					onChange={ () => null }
				/>,
			},
			callback: ( wrapper, index ) => {
				expect( wrapper.find( '.test4-class' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.test4-class button' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.test4-class button' ).hostNodes().prop( 'disabled' ) ).toBeFalsy();
				expect( wrapper.find( '.test4-class button.is-active' ).hostNodes() ).toHaveLength( 0 );
				expect( wrapper.find( '.test4-class button .components-dropdown-button__indicator' ).hostNodes() ).toHaveLength( 1 );
				expect( wrapper.find( '.components-color-palette' ).hostNodes() ).toHaveLength( 0 );

				{
					wrapper.find( '.test4-class .components-dropdown-button__toggle' ).hostNodes().simulate( 'click' );
					expect( toJson( wrapper, { mode: 'deep' } ) ).toMatchSnapshot( getSnapshotName( 'open', index ) );
					expect( wrapper.find( '.components-color-palette' ).hostNodes() ).toHaveLength( 1 );

					const outerNode = document.createElement( 'div' );
					document.body.appendChild( outerNode );
					outerNode.dispatchEvent( new Event( 'click' ) );
					wrapper.update();

					expect( wrapper.find( '.components-color-palette' ).hostNodes() ).toHaveLength( 0 );
				}

				{
					wrapper.find( '.test4-class .components-dropdown-button__toggle' ).hostNodes().simulate( 'click' );
					expect( wrapper.find( '.components-color-palette__custom-color' ).hostNodes() ).toHaveLength( 1 );
					expect( wrapper.find( '.components-color-picker' ).hostNodes() ).toHaveLength( 0 );

					wrapper.find( '.components-color-palette__custom-color .components-button' ).hostNodes().simulate( 'click' );
					expect( toJson( wrapper, { mode: 'deep' } ) ).toMatchSnapshot( getSnapshotName( 'open-color-picker', index ) );
					expect( wrapper.find( '.components-color-picker' ).hostNodes() ).toHaveLength( 1 );
					expect( wrapper.find( '.components-color-picker .components-button' ).hostNodes() ).toHaveLength( 1 );

					document.body.appendChild( wrapper.find( '.components-color-picker' ).hostNodes().instance() );
					const dropdown = wrapper.find( Dropdown ).instance();
					dropdown.closeIfClickOutside( { target: wrapper.find( '.components-color-picker .components-button' ).hostNodes().instance() } );
					wrapper.update();

					expect( wrapper.find( '.components-color-palette__custom-color' ).hostNodes() ).toHaveLength( 1 );

					wrapper.find( '.test4-class .components-dropdown-button__toggle' ).hostNodes().simulate( 'click' );
					expect( wrapper.find( '.components-color-palette__custom-color' ).hostNodes() ).toHaveLength( 0 );
				}

				{
					wrapper.find( '.test4-class .components-dropdown-button__toggle' ).hostNodes().simulate( 'click' );
					expect( wrapper.find( '.components-color-palette__custom-color' ).hostNodes() ).toHaveLength( 1 );

					wrapper.find( '.components-color-palette__custom-color .components-button' ).hostNodes().simulate( 'click' );
					expect( wrapper.find( '.components-color-picker .components-button' ).hostNodes() ).toHaveLength( 1 );

					const dropdown = wrapper.find( Dropdown ).instance();
					dropdown.closeIfClickOutside( { target: wrapper.find( '.test4-content-class' ).hostNodes().instance() } );
					wrapper.update();

					expect( wrapper.find( '.components-color-palette__custom-color' ).hostNodes() ).toHaveLength( 1 );
				}
			},
		},
	].forEach( ( { props, callback }, index ) => {
		it( `should render DropdownButton ${ index }`, () => {
			const wrapper = mount(
				<DropdownButton { ...props }/>,
			);

			expect( toJson( wrapper, { mode: 'deep' } ) ).toMatchSnapshot( getSnapshotName( 'test', index ) );

			callback( wrapper, index );
		} );
	} );
} );
