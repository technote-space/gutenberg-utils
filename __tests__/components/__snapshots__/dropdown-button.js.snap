// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownButton should render DropdownButton 0: open--0 1`] = `
<div
  className="components-dropdown components-dropdown-toggle test1-class"
>
  <button
    aria-expanded={true}
    aria-haspopup="true"
    aria-label="test1-tooltip"
    aria-pressed={true}
    className="components-button components-dropdown-button__toggle components-toolbar__control utils--components-toolbar__control is-active is-pressed has-icon"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  />
  <span>
    <div
      onKeyDown={[Function]}
      tabIndex="-1"
    >
      <div
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="components-popover components-dropdown__content components-dropdown-button__popover test1-content-class is-without-arrow"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
          >
            <div
              className="components-popover__content"
              tabIndex="-1"
            >
              <div
                className="popover1"
              >
                test1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </span>
</div>
`;

exports[`DropdownButton should render DropdownButton 0: test--0 1`] = `
<div
  className="components-dropdown components-dropdown-toggle test1-class"
>
  <button
    aria-expanded={false}
    aria-haspopup="true"
    aria-label="test1-tooltip"
    aria-pressed={true}
    className="components-button components-dropdown-button__toggle components-toolbar__control utils--components-toolbar__control is-active is-pressed has-icon"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  />
</div>
`;

exports[`DropdownButton should render DropdownButton 1: test--1 1`] = `
<div
  className="components-dropdown components-dropdown-toggle test2-class"
>
  <button
    aria-expanded={false}
    aria-haspopup="true"
    aria-label="test2-label"
    aria-pressed={true}
    className="components-button components-dropdown-button__toggle components-toolbar__control utils--components-toolbar__control is-active is-pressed has-icon"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="dashicon dashicons-menu"
      focusable={false}
      height={20}
      role="img"
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17 7V5H3v2h14zm0 4V9H3v2h14zm0 4v-2H3v2h14z"
      />
    </svg>
  </button>
</div>
`;

exports[`DropdownButton should render DropdownButton 2: test--2 1`] = `
<div
  className="components-dropdown components-dropdown-toggle test3-class"
>
  <button
    aria-expanded={false}
    aria-haspopup="true"
    aria-label="test3-label"
    aria-pressed={false}
    className="components-button components-dropdown-button__toggle components-toolbar__control utils--components-toolbar__control has-text has-icon"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="dashicon dashicons-menu"
      focusable={false}
      height={20}
      role="img"
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17 7V5H3v2h14zm0 4V9H3v2h14zm0 4v-2H3v2h14z"
      />
    </svg>
    <span
      className="components-dropdown-menu__indicator"
    />
  </button>
</div>
`;
